`phpstan` failing at nightly build

### Description

`phpstan` pipeline is failing on nightly build and seems to be a hard failure. Needs some investigation regarding it.

```
Line   lib/Crypto/Encryption.php                                           
 ------ -------------------------------------------------------------------- 
  321    Comparison operation ">" between 0 and 0 is always false.           
         💡 Because the type is coming from a PHPDoc, you can turn off this   
            check by setting treatPhpDocTypesAsCertain: false in your        
            ./phpstan.neon.                                                  
 ------ -------------------------------------------------------------------- 


 [ERROR] Found 1 error                                                          

make: *** [Makefile:112: test-php-phpstan] Error 1
```

Failed Build: https://drone.owncloud.com/owncloud/encryption/2914/2/4

--- a/composer.json
+++ b/composer.json
@@ -11,7 +11,7 @@
   "require": {
   },
   "require-dev": {
-    "bamarni/composer-bin-plugin": "^1.4"
+    "bamarni/composer-bin-plugin": "^1.8"
   },
   "extra": {
     "bamarni-bin": {

--- a/vendor-bin/phpstan/composer.json
+++ b/vendor-bin/phpstan/composer.json
@@ -1,5 +1,5 @@
 {
     "require": {
-        "phpstan/phpstan": "1.9.18"
+        "phpstan/phpstan": "^1.10.3"
     }
 }

